src/main/java/formflow/library/config/ActionManager.java

```diff
@@ -166,14 +166,15 @@ public void handleBeforeDisplayAction(ScreenNavigationConfiguration currentScree
    *
    * @param currentScreen  The screen that we are currently saving data from.
    * @param formSubmission The current form submission
+   * @param submission     The submission object after changes to the current screen have been saved to the repository
```

Comment:
Validation is run before we merge the data into the Submission.  So the Submission does not have the currently submitted FormSubmission data yet.   The current Submission here is what is currently in the DB _before_ the current screens data has been merged into it.   That's my understanding at least, is yours different?

Reply:
Yes, you are totally right! Will update.